linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -1063,22 +1084,27 @@ public void close() {
       final Enumerable<TSource> outer, final Enumerable<TInner> inner,
       final Function1<TSource, TKey> outerKeySelector,
       final Function1<TInner, TKey> innerKeySelector,
+      final Predicate2<TSource, TInner> predicate,
       final Function2<TSource, TInner, TResult> resultSelector,
       final EqualityComparer<TKey> comparer, final boolean generateNullsOnLeft,
       final boolean generateNullsOnRight) {
     return new AbstractEnumerable<TResult>() {
       public Enumerator<TResult> enumerator() {
+        final Enumerable<TInner> innerToLookUp = generateNullsOnLeft
```

Comment:
@rubenada ,I introduce a innerToLookUp to refer the inner , if current join is a right join, we should
figure out the right list first, if not, then keep the original inner here.

Reply:
thanks @LaiZhou; since the code is, in my opinion, not so straightforward to comprehend, maybe a small comment (like the one you just wrote above) will help future readers to understand the logic behind.

Reply:
The comment is added now.

Reply:
@LaiZhou , sorry, long time since I last checked this code. Could you please remind me why if the join is a RIGHT (or FULL, I would add) we "should figure out the right list first" ?